src/ui/src/components/core/content/CoreChatBot/CoreChatbotMessage.vue

```diff
@@ -7,10 +7,58 @@
 		>
 			<WdsLoaderDots v-if="displayLoader" />
 			<div v-else class="CoreChatbotMessage__content__text">
-				<BaseMarkdown v-if="useMarkdown" :raw-text="content">
-				</BaseMarkdown>
+				<!-- Handle multimodal content (array) -->
+				<template v-if="Array.isArray(content)">
+					<!-- Render text fragments first -->
+					<template
+						v-for="(fragment, index) in content"
+						:key="`text-${index}`"
+					>
+						<div
+							v-if="fragment.type === 'text'"
+							class="content-fragment"
+						>
+							<BaseMarkdown
+								v-if="useMarkdown"
+								:raw-text="fragment.text || ''"
+							>
+							</BaseMarkdown>
+							<span v-else>{{ fragment.text }}</span>
+						</div>
+					</template>
+
+					<!-- Render images in horizontal scrollable container -->
+					<div
+						v-if="hasImages(content)"
+						class="message-images-wrapper"
+					>
+						<div class="message-images-container">
+							<template
+								v-for="(fragment, index) in content"
+								:key="`image-${index}`"
+							>
+								<img
+									v-if="fragment.type === 'image_url'"
+									:src="fragment.image_url?.url"
+									:alt="'Image from conversation'"
+									class="message-image-preview"
+									loading="lazy"
+									@error="handleImageError"
+									@click="
+										openImageModal(fragment.image_url?.url)
+									"
```

Comment:
1. I guess we don't want to display the image if we don't have any URL.
2. You can simply use `modalImageUrl` (see my next comments).

```suggestion
									v-if="fragment.type === 'image_url' && fragment.image_url?.url"
									:src="fragment.image_url.url"
									alt="Image from conversation"
									class="message-image-preview"
									loading="lazy"
									@error="handleImageError"
									@click="modalImageUrl = fragment.image_url.url"
```